src/components/general/DatePicker.js

```diff
@@ -3,195 +3,132 @@ import PropTypes from "prop-types";
-				day,
-				year
-			} = item;
-			this.props.onSelect(`${year}-${month}-${day}`);
-		}
+	/** Returns an array of days in the current state's month when called */
+	getDayArray() {
+		const { month, day, year } = this.state;
 
-		clickActionMonth(item) {
-			const {
-				day,
-				year
-			} = this.state;
-
-			item = this.prepend0(item);
-
-			this.setState({ month: item });
-			const month30 = [false, false, false, true, false, true,
-				false, false, true, false, true, false];
-
-			if (item === 2)
-				if (year != 0 && this.leapYear(year)) {
-					this.setState({ dayArr: Array.from({ length: 29 }, (v, k) => k + 1) });
-					if (day > 29) this.setState({ day: 29 });
-				}
-				else {
-					this.setState({ dayArr: Array.from({ length: 28 }, (v, k) => k + 1) });
-					if (day > 28) this.setState({ day: 28 });
-				}
-			if (month30[item]) {
-				this.setState({ dayArr: Array.from({ length: 30 }, (v, k) => k + 1) });
-				if (day > 30) this.setState({ day: 30 });
-			}
-			else if (item !== 2) {
-				this.setState({ dayArr: Array.from({ length: 31 }, (v, k) => k + 1) });
-			}
-			if (day !== "" && year !== "")
-				this.update({ day: day, month: item, year: year });
-		}
-		clickActionDay(item) {
-			const {
-				month,
-				year
-			} = this.state;
+		const daysInMonth = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+		const monthInt = parseInt(month);
 
-			item = this.prepend0(item);
-			this.setState({ day: item });
+		// If the state's day is more than the max days in a month, set the day to the max day
+		const setMaxDay = (maxDay) => { if (day > maxDay) this.setState({ day: maxDay }); };
 
-			if (month !== "" && year !== "")
-				this.update({ day: item, month: month, year: year });
+		if (monthInt == 2 && this.isLeapYear(year)) {
+			setMaxDay(29);
+			return Array.from({ length: 29 }, (v, k) => k + 1);
 		}
+		setMaxDay(daysInMonth[monthInt - 1]);
+		return Array.from({ length: daysInMonth[monthInt - 1] }, (v, k) => k + 1);
+	}
 
-		clickActionYear(item) {
-			const {
-				month,
-				day
-			} = this.state;
+	/**
+	 * Returns true if the input year is a leap year. Must be an int.
+	 * @param {Number} year Any year such as 2020 or 101
+	*/
+	isLeapYear(year) { return (year % 4 === 0) && (year % 100 != 0) || (year % 400 == 0) }
 
-			this.setState({ year: item });
+	/** Sets the date by calling the onSelect function in the Form component. */
+	setDate(item) {
+		const { month, day, year } = item;
+		this.props.onSelect(`${year}-${month}-${day}`);
+	}
 
-			if (month === "2" && this.leapYear(item)) {
-				this.setState({ dayArr: Array.from({ length: 29 }, (v, k) => k + 1) });
-				if (day > 29) this.setState({ day: "29" });
-			}
-			if (month !== "" && day !== "")
-				this.update({ day: day, month: month, year: item });
+	/**
+	 * Calls the setDate function and sets the month/day/year based on a given value and type.
+	 * @param {Number} item Any number.
+	 * @param {String} type A string that is "month", "day", "year".
```

Comment:
The description should be what the value holds. If it can only be a certain value then write something like.
" String representating the value being changed. Must be one of these values. ["month", "day", "year"]`"

Reply:
The item should be something like "Number representing the value to update the state to"